### ISSUE TITLE
Implement support for webstrates larger than 16megs

### ISSUE THREAD

**Issue by cklokmose** [cklokmose/Webstrates#7]
Currently Webstrates uses MongoDB as its database backend. 
MongoDB has a limitation of 16 MB per document.
Investigate wether this limitation can be overcome e.g. by using the GridFS API.
https://docs.mongodb.org/manual/core/gridfs/

**Comment by raedle** [comment]
GridFS could be a way to go since Webstrates already uses MongoDB as data backend.

"GridFS is a specification for storing and retrieving files that exceed the BSON-document size limit of 16 MB."

https://docs.mongodb.com/manual/core/gridfs/

**Comment by kbadk** [comment]
[GridFS](https://docs.mongodb.com/manual/core/gridfs/) is a way of storing files in the MongoDB database. These files are just that – files. It's not possible to query or perform operations on the file, anymore than it is possible to do so with files stored manually in the file system. As a result, GridFS wouldn't be fit for string webstrate documents.

The only solution I can think of would be to split large documents into multiple objects, but I think it would be a struggle to get ShareDB to play nice with this.

**Comment by raedle** [comment]
Can sharedb cope with docs larger than 16MB and is it only due to Webstrates internal db structure?

If yes, then a solution could be to split webstrate documents into chunks and set refs in the tree to sub documents. The server would have to build the whole tree using refs.

This would eventually allow for partial transclusion.

**Comment by kbadk** [comment]
ShareDB doesn't support this. It may be possible to get ShareDB to work with multipl documents with modifications – as mentioned above – but it's not something I've had time to look too much into just yet.

**Comment by curran** [comment]
I would agree, 16MB seems too big for the browser anyway. Interesting discussion!
